app/src/main/java/org/hwyl/sexytopo/control/graph/GraphView.java

```diff
@@ -170,11 +180,31 @@ public void initialisePaint() {
         highlightPaint.setStrokeWidth(HIGHLIGHT_OUTLINE);
         highlightPaint.setColor(HIGHLIGHT_COLOUR.intValue);
 
-        legPaint.setARGB(127, 255, 0, 0);
+        // active legs/splays
         int legStrokeWidth = PreferenceAccess.getInt(getContext(), "pref_leg_width", 3);
         legPaint.setStrokeWidth(legStrokeWidth);
         legPaint.setColor(LEG_COLOUR.intValue);
 
+        latestLegPaint.setStrokeWidth(legStrokeWidth);
+        latestLegPaint.setColor(LATEST_LEG_COLOUR.intValue);
+
+        int splayStrokeWidth = PreferenceAccess.getInt(getContext(), "pref_splay_width", 1);
```

Comment:
Setting this here is probably a good idea rather than in the draw thread, but it does mean changing this setting wouldn't update the display. This change needs an `initialisePaint()` in `onResume()` I think.

Reply:
It's called here:
```
    private void syncGraphWithSurvey() {
        Survey survey = getSurvey();
        graphView.initialisePaint();
        graphView.setProjectionType(getProjectionType());
        graphView.setProjection(getProjection(survey));
        graphView.setSurvey(survey);
        graphView.setSketch(getSketch(survey));

        double surveyLength = SurveyStats.calcTotalLength(survey);
        double surveyHeight = SurveyStats.calcHeightRange(survey);
        graphView.setCachedStats(surveyLength, surveyHeight);

        graphView.invalidate();
    }
```
I've tested it and they do update when you press back from the settings page

Reply:
Ah, fair.